metrics-report-generator/src/main/scala/it/pagopa/interop/metricsreportgenerator/util/Jobs.scala

```diff
@@ -43,6 +45,51 @@ class Jobs(config: Configuration, readModel: ReadModelService)(implicit
       }
   }
 
+  def getAgreementSheet(implicit ec: ExecutionContext): Future[Sheet] = {
+    logger.info("Gathering Agreements Information")
+    val headerList = List(
+      "EserviceId",
+      "Eservice",
+      "Producer",
+      "ProducerId",
+      "Consumer",
+      "ConsumerId",
+      "AgreementId",
+      "Purposes",
+      "PurposeIds"
+    )
+    val headerRow  =
+      Row(style = SheetStyle.headerStyle).withCellValues(headerList)
+
+    val columns = (0 until (headerList.size)).toList
+      .map(index => Column(index = index, style = CellStyle(font = Font(bold = true)), autoSized = true))
+
+    ReadModelQueries
+      .getAllActiveAgreements(config.collections, readModel)
+      .zip(ReadModelQueries.getAllPurposes(config.collections, readModel))
```

Comment:
considering the memory issues, let's try to easy the load on the db by avoiding parallel queries, using a `flatMap` instead of `zip`